verify-rpms/rpm_verifier.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+from typing import Callable
+
+import click
+
+
+def get_rpmdb(container_image: str) -> Path:
+    raise NotImplementedError()
+
+
+def get_rpms(rpmdb: Path) -> list[str]:
+    raise NotImplementedError()
+
+
+def get_unsigned_rpms(rpms: list[str], verifier: Callable[[str], bool]) -> list[str]:
```

Comment:
I don't think its really useful to make the verifier injectable in this case

Reply:
How come? It will allow us to easily separate in tests the functionality of returning the list of unsigned rpms from the functionality of figuring out whether an individual rpm is signed.

Reply:
because this isn't really a complex or useful logic that we would need to test, its a single Python comprehension statement.

Reply:
With your permission, I'll keep it injectable. It makes things simpler for me.

Reply:
it hides away the true nature if what is going on - the thing we extract from the RPM DB is not a list of RPMs, its a list of records about the RPMs where each record contains a field which indicats of the RPM is signed or not.

Reply:
Is this about the name of the variable or having the verifier injected?

Reply:
having an injected verifier evokes the "strategy" pattern. It implies you can have different verifiers for different cases, which you can't really because the DB extraction command is tailored for inspecting signature status. 

In effect it tries to create a boundary between pieces of code that are tightly rather then loosely coupled.

Reply:
The reason for having things here injected is to make tests easier rather than for allowing different plugable implementations.

The thing I'm worried about with the implementation of this module is having to write larger test cases in which we will have to monkeypatch multiple subprocess calls and verify the comprehension logics. Having more fine-grained function and injecting the dependencies will allow us to avoid much of that (I realized now that I didn't make the external call injectable).

Reply:
80% of this code needs a single test: run on a given well-known image and inspect the results. Its a two line test, with possibly an extra dockerfile for building the test image.

Having finer grained test for this will just hardwire us into a particular implementation. Especially if we worry about the subprocess invocations too much and try to factor then out. Consider that you woulnd't be worrying about this if the code was in bash of we would be using the RPM Python library.

Reply:
If it was written in bash, I'd be worried that it's not working and I'm unable to test it.

> 80% of this code needs a single test: run on a given well-known image and inspect the results. Its a two line test, with possibly an extra dockerfile for building the test image.

We must be talking about different things. Are you talking about skipping unit tests altogether and just implementing integration tests?

Reply:
Either way, I'm not comfortable with writing the code this way.
Seeing that I'll be one of the few people that will be here around the year end, I'd like to have the code in such structure that is maintainable for me.